spring-boot-modules/spring-boot-3/src/main/java/com/baeldung/sample/singleton/SingletonPattern.java

```diff
@@ -0,0 +1,24 @@
+package com.baeldung.sample.singleton;
+
+public final class SingletonPattern {
```

Comment:
The class name - `SingletonPattern` - might be worth improving:

`ThreadSafeSingleInstance`?

Reply:
Makes sense, fixed.